src/model.rs

```diff
@@ -248,6 +248,14 @@ impl ServerInfo {
 			permissions: try!(remove(&mut value, "permissions").and_then(Permissions::decode)),
 		})
 	}
+
+	/// Returns the formatted URL of the server's icon.
+	///
+	/// Returns None if the server does not have an icon.
+	pub fn icon_url(&self) -> Option<String> {
+		self.icon.as_ref().and_then(|icon|
+			Some(format!(api_concat!("/guilds/{}/icons/{}.jpg"), self.id, icon)))
```

Comment:
No need to write `.and_then(|_| Some(...))` - using `.map(|_| ...)` should be fine.